File: frontend/src/types/notification.ts
export type NotificationType =
  | 'upload_success'
  | 'upload_failed'
  | 'platform_published'
  | 'platform_failed'
  | 'schedule_reminder'
  | 'schedule_completed'
  | 'token_expiring'
  | 'token_expired'
  | 'credit_low'
  | 'credit_depleted'
  | 'milestone'
  | 'report_ready'
  | 'payment_reminder'
  | 'payment_failed'

export type NotificationCategory = 'upload' | 'schedule' | 'channel' | 'ai' | 'analytics' | 'subscription'

export interface Notification {
  id: number
  type: NotificationType
  category: NotificationCategory
  title: string
  message: string
  isRead: boolean
  referenceType?: string // 'video' | 'schedule' | 'channel' | 'credit' | 'report' | 'payment'
  referenceId?: number
  createdAt: string
}

export interface NotificationSetting {
  category: NotificationCategory
  inApp: boolean
  email: boolean
  kakao: boolean
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/notification/Notification.kt
package com.ongo.domain.notification

import com.ongo.common.enums.NotificationType
import java.time.LocalDateTime

data class Notification(
    val id: Long? = null,
    val userId: Long,
    val type: NotificationType,
    val title: String,
    val message: String,
    val isRead: Boolean = false,
    val referenceType: String? = null,
    val referenceId: Long? = null,
    val createdAt: LocalDateTime? = null,
)
